### PR TITLE
[cloud_firestore] convert tests to e2e

### PR DESCRIPTION
**Pull request by cyanglaz**
## Description

Converting tests in the plugin to use e2e test.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] If the pull request affects only one plugin, the PR title starts with the name of the plugin in brackets (e.g. [cloud_firestore])
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/cloud_firestore/cloud_firestore/example/test_driver/cloud_firestore_e2e.dart [hunk 1]
```diff
@@ -1,12 +1,549 @@
...
+import 'package:firebase_core/firebase_core.dart';
 import 'package:cloud_firestore/cloud_firestore.dart';
 
 void main() {
   E2EWidgetsFlutterBinding.ensureInitialized();
 
-  testWidgets('Can initialize the plugin', (WidgetTester tester) async {
-    Firestore firestore = Firestore();
-    expect(firestore, isNotNull);
+  group('$Firestore', () {
```
Comment (blasten) [inline packages/cloud_firestore/cloud_firestore/example/test_driver/cloud_firestore_e2e.dart:11]: nit: is this meant to be `$Firestore` or `Firestore`?

File: packages/cloud_firestore/cloud_firestore/example/test_driver/cloud_firestore_e2e.dart [hunk 2]
```diff
@@ -1,12 +1,549 @@
...
+          .collection('messages')
+          .where('message', isEqualTo: 'Hello world!')
+          .limit(1);
+      final QuerySnapshot querySnapshot = await query.getDocuments();
+      expect(querySnapshot.metadata, isNotNull);
+      expect(querySnapshot.documents.first['message'], 'Hello world!');
+      final DocumentReference firstDoc =
+          querySnapshot.documents.first.reference;
+      final DocumentSnapshot documentSnapshot = await firstDoc.get();
+      expect(documentSnapshot.data['message'], 'Hello world!');
```
Comment (blasten) [inline packages/cloud_firestore/cloud_firestore/example/test_driver/cloud_firestore_e2e.dart:57]: subjective nit: empty line between `action/assert` blocks for better readability.

Comment (blasten) [reply packages/cloud_firestore/cloud_firestore/example/test_driver/cloud_firestore_e2e.dart:57]: the ones below too. e.g.

```dart
final DocumentSnapshot cachedSnapshot = await firstDoc.get(source: Source.cache);
expect(cachedSnapshot.data['message'], 'Hello world!');

final DocumentSnapshot snapshot = await firstDoc.snapshots().first;
expect(snapshot.data['message'], 'Hello world!');
```

**Comment by cyanglaz** [comment]
Closing as this doesn't pass the firebase tests. I don't have time to work on this in a near future. Might reopen if i figured it out.

**Pull request closed without merging**